components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/transformations/DiscardedRowsFilter.java

```diff
@@ -41,11 +41,13 @@ public Boolean call(String[] tokens) throws Exception {
         try {
             Boolean keep = true;
             for (Integer index : indices) {
-                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) || MLConstants.NA.equals(tokens[index])) {
+                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) ||
```

Comment:
Using an Enum is crucial for maintenance of this code, since the same logic is there in multiple places. If isValidEnum won't do, please use the approach you suggested.